src/collections/vec.rs

```diff
@@ -1806,35 +1837,79 @@ impl<'bump, T: 'bump + Copy> Vec<'bump, T> {
     /// assert_eq!(vec, "Hello, world!".as_bytes());
     /// ```
     ///
-    /// [`extend`]: #method.extend_from_slice
+    /// [`extend_from_slice`]: #method.extend_from_slice
+    /// [`extend_from_slices`]: #method.extend_from_slices
     pub fn extend_from_slice_copy(&mut self, other: &[T]) {
-
         // Reserve space in the Vec for the values to be added
         let old_len = self.len();
-        self.reserve(other.len());
-
         let new_len = old_len + other.len();
-        debug_assert!(new_len <= self.capacity());
+        self.reserve(other.len());
 
         // Copy values into the space that was just reserved
         // SAFETY:
-        // * `src` is valid for reads of `other.len()` values by virtue of being a `&[T]`.
+        // * `self` has enough capacity to store `other.len()` more items as `self.reserve(other.len())`
+        //   above guarantees that.
+        // * Source and destination data ranges cannot overlap as we just reserved the destination
+        //   range from the bump.
+
+        unsafe {
+            self.extend_from_slice_copy_unchecked(other);
+            // The elements at `old_len..new_len` are initialized by `extend_from_slice_copy_unchecked`
+            // above. Update the Vec's length to reflect the number of items added.
+            self.set_len(new_len);
+        }
+    }
+
+    /// For each slice in `slices`, copies all elements in the slice and appends them to the `Vec`.
+    ///
+    /// This method is equivalent to calling [`extend_from_slice_copy`] in a loop, but is able
+    /// to precompute the total amount of space to reserve in advance. This reduces the potential
+    /// maximum number of reallocations needed from one-per-slice to just one.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bumpalo::{Bump, collections::Vec};
+    ///
+    /// let b = Bump::new();
+    ///
+    /// let mut vec = bumpalo::vec![in &b; 1];
+    /// vec.extend_from_slices_copy(&[&[2, 3], &[], &[4]]);
+    /// assert_eq!(vec, [1, 2, 3, 4]);
+    /// ```
+    ///
+    /// ```
+    /// use bumpalo::{Bump, collections::Vec};
+    ///
+    /// let b = Bump::new();
+    ///
+    /// let mut vec = bumpalo::vec![in &b; 'H' as u8];
+    /// vec.extend_from_slices_copy(&["ello,".as_bytes(), &[], " world!".as_bytes()]);
+    /// assert_eq!(vec, "Hello, world!".as_bytes());
+    /// ```
+    ///
+    /// [`extend_from_slice_copy`]: #method.extend_from_slice_copy
+    pub fn extend_from_slices_copy(&mut self, slices: &[&[T]]) {
+        // Reserve the total amount of capacity we'll need to safely append the aggregated contents
+        // of each slice in `slices`.
+        let bytes_to_reserve: usize = slices.iter().map(|buf| buf.len()).sum();
+        self.reserve(bytes_to_reserve);
```

Comment:
This is not actually bytes, its elements, right? Should update the variable names accordingly.